hack/update-licenses.sh

```diff
@@ -0,0 +1,186 @@
+      LICENSE) find_names=(-iname 'licen[sc]e*')
+               find_maxdepth=1
+               # Sadly inconsistent in the wild, but mostly license files
+               # containing copyrights, but no readme/notice files containing
+               # licenses (except to "see license file")
+               ensure_pattern="license|copyright"
+               ;;
+    # We search READMEs for copyrights and this includes notice files as well
+    # Look in as many places as we find files matching
+    COPYRIGHT) find_names=(-iname 'notice*' -o -iname 'readme*')
+               find_maxdepth=3
+               ensure_pattern="copyright"
+               ;;
+  esac
+
+  # Start search at package root
+  case ${package} in
+    github.com/*|golang.org/*|bitbucket.org/*)
+     package_root=$(echo ${package} |awk -F/ '{ print $1"/"$2"/"$3 }')
+     ;;
+    go4.org/*)
+     package_root=$(echo ${package} |awk -F/ '{ print $1 }')
+     ;;
+    *)
+     package_root=$(echo ${package} |awk -F/ '{ print $1"/"$2 }')
+     ;;
+  esac
+
+  # Find files - only root and package level
+  local_files=($(
+    for dir_root in ${package} ${package_root}; do
+      [[ -d ${DEPS_DIR}/${dir_root} ]] || continue
+
+      # One (set) of these is fine
+      find ${DEPS_DIR}/${dir_root} \
+          -xdev -follow -maxdepth ${find_maxdepth} \
+          -type f "${find_names[@]}"
+    done | sort -u))
+
+  local index
+  local f
+  index="${package}-${type}"
+  if [[ -z "${CONTENT[${index}]-}" ]]; then
+    for f in ${local_files[@]-}; do
+      # Find some copyright info in any file and break
+      if egrep -i -wq "${ensure_pattern}" "${f}"; then
+        CONTENT[${index}]="${f}"
+        break
+      fi
+    done
+  fi
+}
+
+
+#############################################################################
+# MAIN
+#############################################################################
+
+# Check bash version
+if ((${BASH_VERSINFO[0]}<4)); then
+  echo
+  echo "ERROR: Bash v4+ required."
+  # Extra help for OSX
+  if [[ "$(uname -s)" == "Darwin" ]]; then
+    echo
+    echo "Ensure you are up to date on the following packages:"
+    echo "$ brew install md5sha1sum bash"
+  fi
+  echo
+fi
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+LICENSES_FILE="${REPO_ROOT}/LICENSES"
+TMP_LICENSE_FILE="/tmp/cluster-registry.LICENSES.$$"
+DEPS_DIR="vendor"
+
+cd "${REPO_ROOT}"
+declare -Ag CONTENT
+
+# Put the cluster-registry LICENSE on top
+(
+echo "================================================================================"
+echo "= cluster-registry licensed under: ="
+echo
+cat "${REPO_ROOT}/LICENSE"
+echo
+echo "= LICENSE $(cat "${REPO_ROOT}/LICENSE" | md5sum | awk '{print $1}')"
+echo "================================================================================"
+) > "${TMP_LICENSE_FILE}"
+
+# Loop through every package in vendor.
+for PACKAGE in $(dep status | \
+                 tail -n +2 |
+                 awk '{print $1}' |
+                 sort -f); do
```

Comment:
Hmm. I think you're right. Since `dep` isn't finalized, though, and there's nothing in the documentation I found that says that the sorted order is guaranteed, I'd rather be safe and sort again.